chain/indexer/src/handlers.rs

```diff
@@ -1,45 +1,72 @@
     hopr_node_safe_registry::HoprNodeSafeRegistryEvents, hopr_ticket_price_oracle::HoprTicketPriceOracleEvents,
     hopr_token::HoprTokenEvents,
 };
-use chain_db::traits::HoprCoreEthereumDbActions;
 use chain_types::chain_events::{ChainEventType, NetworkRegistryStatus};
 use chain_types::ContractAddresses;
 use ethers::{contract::EthLogDecode, core::abi::RawLog};
+use futures::TryStreamExt;
+use hopr_crypto_types::keypairs::ChainKeypair;
+use hopr_crypto_types::prelude::{Hash, Keypair};
 use hopr_crypto_types::types::OffchainSignature;
+use hopr_db_api::errors::DbError;
+use hopr_db_api::errors::DbError::LogicalError;
+use hopr_db_api::{HoprDbAllOperations, OpenTransaction, SINGULAR_TABLE_FIXED_ID};
+use hopr_db_entity::conversions::channels::ChannelStatusUpdate;
+use hopr_db_entity::conversions::tickets::model_to_acknowledged_ticket;
+use hopr_db_entity::{account, chain_info, channel, ticket};
 use hopr_internal_types::prelude::*;
 use hopr_primitive_types::prelude::*;
-use multiaddr::Multiaddr;
+use sea_orm::{ActiveModelTrait, ColumnTrait, EntityTrait, IntoActiveModel, QueryFilter, Set};
+use std::fmt::Formatter;
 use std::ops::{Add, Sub};
 use std::time::{Duration, SystemTime};
-use std::{str::FromStr, sync::Arc};
-use tracing::{error, info, trace, warn};
+use tracing::{debug, error, info, trace, warn};
 
 /// Event handling object for on-chain operations
 ///
 /// Once an on-chain operation is recorded by the [crate::block::Indexer], it is pre-processed
 /// and passed on to this object that handles event specific actions for each on-chain operation.
 ///
-#[derive(Debug, Clone)]
-pub struct ContractEventHandlers<U: HoprCoreEthereumDbActions + std::fmt::Debug> {
+#[derive(Clone)]
+pub struct ContractEventHandlers<Db: Clone> {
     /// channels, announcements, network_registry, token: contract addresses
     /// whose event we process
     addresses: ContractAddresses,
     /// Safe on-chain address which we are monitoring
     safe_address: Address,
     /// own address, aka message sender
-    chain_key: Address,
+    chain_key: ChainKeypair, // TODO: store only address here once Ticket have TryFrom DB
     /// callbacks to inform other modules
-    db: Arc<RwLock<U>>,
+    db: Db,
 }
 
-impl<U: HoprCoreEthereumDbActions + std::fmt::Debug> ContractEventHandlers<U> {
-    pub fn new(addresses: ContractAddresses, safe_address: Address, chain_key: Address, db: Arc<RwLock<U>>) -> Self {
+impl<Db: Clone> std::fmt::Debug for ContractEventHandlers<Db> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ContractEventHandler")
+            .field("addresses", &self.addresses)
+            .field("safe_address", &self.safe_address)
+            .field("chain_key", &self.chain_key)
+            .finish_non_exhaustive()
+    }
+}
+
+async fn channel_model_from_id(tx: &OpenTransaction, channel_id: Hash) -> Result<Option<channel::Model>> {
```

Comment:
These functions would ideally be passed on to the DB API.

Reply:
The problem is that here we need the `Model` almost always, otherwise we would convert back and forth from the business objects.